code/modules/clothing/gloves/_gloves.dm

```diff
@@ -11,13 +11,21 @@
 	strip_delay = 20
 	equip_delay_other = 40
 
+/obj/item/clothing/gloves/Initialize()
+	. = ..()
+	AddComponent(/datum/component/redirect, list(COMSIG_COMPONENT_CLEAN_ACT), CALLBACK(src, .proc/clean_blood))
+
+/obj/item/clothing/gloves/proc/clean_blood(strength)
+	if(strength < CLEAN_STRENGTH_BLOOD)
+		return
+	transfer_blood = 0
 
 /obj/item/clothing/gloves/worn_overlays(isinhands = FALSE)
 	. = list()
 	if(!isinhands)
 		if(damaged_clothes)
 			. += mutable_appearance('icons/effects/item_damage.dmi', "damagedgloves")
-		if(blood_DNA)
+		if(SendSignal(COMSIG_FORENSICS_HAS_EVIDENCE) & FORENSICS_COMPONENT_HAS_BLOOD_DNA)

```

Comment:
All these send signals should just be getcomponent followed by querying the component for whatever it wants